packages/framework/tinylicious-client/src/test/TinyliciousClient.spec.ts

```diff
@@ -277,4 +279,63 @@ describe("TinyliciousClient", () => {
         await corruptedAliasOp(runtimeOf(do1), "alias");
         assert(await dataCorruption);
     });
+
+    /**
+     * Scenario: Test if TinyliciousClient with forceWriteMode set to true forces the container to write mode.
+     * Keep the original behaviour otherwise.
+     *
+     * Expected behavior: Setting forceWriteMode to true in TinyliciousClientProps should force the container's
+     * connectionMode to `write` while keeping the original behavior if forceWriteMode is not set to true.
+     */
+    it("forces the container to connect with write mode when forceWriteMode is set to true ", async () => {
+        const getConnectionModeFromTinyliciousClient = async (
+            props?: TinyliciousClientProps,
+        ): Promise<ConnectionMode> => {
+            const client = new TinyliciousClient(props);
+            const { container } = await client.createContainer(schema);
+            await container.attach();
+
+            const map1 = container.initialObjects.map1 as SharedMap;
+            map1.set("newpair-id", "test");
+
+            await timeoutPromise((resolve) => container.once("connected", resolve), {
+                durationMs: 1000,
+                errorMsg: "container connect() timeout",
+            });
+
+            return (container as any).container.connectionMode as ConnectionMode;
+        };
+
+        // test forced write mode case
+        const forcedWriteConnectionMode = await getConnectionModeFromTinyliciousClient({ forceWriteMode: true });
+        assert.strictEqual(
+            forcedWriteConnectionMode,
+            "write",
+            "Container does not start with write mode by defaut when forceWriteMode set to true",
+        );
+
+        // test default case
+        const defaultConnectionMode = await getConnectionModeFromTinyliciousClient();
+        assert.strictEqual(
+            defaultConnectionMode,
+            "read",
+            "Container does not start with write mode by defaut when forceWriteMode set to true",
+        );
+
+        // test false case
+        const falseConnectionMode = await getConnectionModeFromTinyliciousClient({ forceWriteMode: false });
+        assert.strictEqual(
+            falseConnectionMode,
+            "read",
+            "Container does not start with write mode by defaut when forceWriteMode set to true",
+        );
+
+        // test undefined case
+        const undefinedConnectionMode = await getConnectionModeFromTinyliciousClient({ forceWriteMode: undefined });
+        assert.strictEqual(
+            undefinedConnectionMode,
+            "read",
+            "Container does not start with write mode by defaut when forceWriteMode set to true",
+        );
```

Comment:
Are you able to test against permissions granted by token?

Reply:
I'm able to test it by manually by removing ScopeType.DocWrite from line 42 in InsecureTinyliciousTokenProvider, but I'm not able to remove the permission in the test, I've tried a few ways and closest I got was these, but they don't work either; ```(container as any).container.scopes = ["doc:read","doc:write","summary:write"];``` or modifying ``(client as any).documentServiceFactory.tokenProvider```

Reply:
Is this matter of providing an argument on InsecureTinyliciousTokenProvider that can accept read/write mode?

Reply:
I found a way to override scopes, but I don't like creating a new property on TinyliciousClientProps. I have committed my changes, but is there a way to do it without creating a new property on TinyliciousClientProps?

Reply:
To test it you could pass (optional) token provider to Tinylicious Client (similar to what we do with azure client).

Reply:
How is my current implementation for read and write permissions?

Reply:
It seems a bit hacky. If anything changes underneath, you will not end up with build-time errors. Option here is:

1. InsecureTinyliciousTokenProvider can accept scopeTypes on constructor.
2. Pass (optional) token provider to Tinylicious Client (similar to what we do with azure client). It seems like a reasonable approach.